packages/web-api/src/WebClient.ts

```diff
@@ -535,6 +545,28 @@ export class WebClient extends Methods {
   // eslint-disable-next-line class-methods-use-this
   private buildResult(response: AxiosResponse): WebAPICallResult {
     let { data } = response;
+    const isGzipResponse = response.headers['content-type'] === 'application/gzip';
+
+    // Check for GZIP response - if so, it is a successful response from admin.analytics.getFile
+    if (isGzipResponse) {
+      // admin.analytics.getFile will return a Buffer that can be unzipped
+      try {
+        const unzippedData = zlib.unzipSync(data).toString().split('\n');
```

Comment:
this is also a great callout! i was originally using the async method `zlib.unzip()` but was running into issues with accessing the file data since it outputs that into a callback and the data couldn't be reached outside the callback 😢 it seemed complex to have all of the remaining logic that would use this file data within the callback. with that being said, if there's another workaround I could use here, i'm definitely open to it! i'll take a stab at converting this to an async function and see what happens 👀